plugins/inputs/diskio/diskio_linux.go

```diff
@@ -15,7 +16,28 @@ type diskInfoCache struct {
 	values       map[string]string
 }
 
-var udevPath = "/run/udev/data"
+var udevPath = []string{"/run/udev/data", "/dev/.udev/db"}
+
+func getUdevDataPath(devName string, major uint64, minor uint64) (out string, err error) {
+	for _, udev := range udevPath {
+		// First try the systemd-path
+		var path = fmt.Sprintf("%s/b%d:%d", udev, major, minor)
+		f, err := os.Open(path)
```

Comment:
To avoid needing to open the file twice, lets have this function return the contents of the file.  It looks like we don't actually need the `udevDataPath` in the `diskInfoCache`.

Reply:
If indeed the `udevDataPath` is not needed inside the `diskInfoCache` then returning the content would indeed be a slightly better flow, I couldn't judge if it was required or not but decided to minimize the changes.